src/pk/ecc/ecc_cmp_oid_str.c

```diff
@@ -0,0 +1,38 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt_private.h"
+
+#ifdef LTC_MECC
+
+/** Compare curve's OID to constant string
+  @param oidstr [in] the desired OID in string format
+  @param key    The ECC key
+  Return        CRYPT_OK if the curve has the correct OID
+*/
+
+int ecc_cmp_oid_str(const char *oidstr, const ecc_key *key)
+{
+   char oid[32];
+   unsigned long oidlen = 32;
+   int err;
+
+   LTC_ARGCHK(key != NULL);
+
+   err = pk_oid_num_to_str(key->dp.oid, key->dp.oidlen, oid, &oidlen);
+   if (err != CRYPT_OK) return err;
+   else if (!XSTRCMP(oidstr, oid)) return CRYPT_OK;
```

Comment:
I prefer to use explicit: `XSTRCMP(x, y) != 0` or `XSTRCMP(x, y) == 0` (applies also to other places)

Reply:
and clang-tidy complains here:
```
src/pk/ecc/ecc_cmp_oid_str.c:30:4: error: do not use 'else' after 'return' [readability-else-after-return,-warnings-as-errors]
   else if (!XSTRCMP(oidstr, oid)) return CRYPT_OK;
   ^~~~~~~~
```

So maybe change it to: `if (XSTRCMP(oidstr, oid) == 0) return CRYPT_OK;`